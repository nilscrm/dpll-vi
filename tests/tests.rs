use insta_cmd::assert_cmd_snapshot;
use rstest::rstest;
use std::{fs::File, path::{Path, PathBuf}, process::Command};

fn test_cnf_files(group_name: &str, path: &Path) {
  insta::with_settings!({
    snapshot_path => format!("snapshots/{}", group_name),
    filters => [(r"Performance\s+Time\s+.*\sms\s+Speed.*IPS", "[Performance...]")] },
  {
    let stdin = File::open(path).unwrap();
    assert_cmd_snapshot!(path.file_stem().unwrap().to_str().unwrap(), Command::new("vine").args(["run", "dpll/main.vi"]).arg("--breadth-first").stdin(stdin));
  });
}

#[rstest]
fn test_minimal(#[files("tests/minimal/*.cnf")] path: PathBuf) {
  test_cnf_files("minimal", &path);
}

#[rstest]
fn test_medium(#[files("tests/medium/*.cnf")] path: PathBuf) {
  test_cnf_files("medium", &path);
}

#[rstest]
fn test_satlib_sat(#[files("tests/uf5-218_first_10/*.cnf")] path: PathBuf) {
  test_cnf_files("uf5-218_first_10", &path);
}

#[rstest]
fn test_satlib_unsat(#[files("tests/uuf50-218_first_10/*.cnf")] path: PathBuf) {
  test_cnf_files("uuf50-218_first_10", &path);
}
