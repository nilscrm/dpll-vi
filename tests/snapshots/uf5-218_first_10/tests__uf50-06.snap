---
source: tests/tests.rs
info:
  program: vine
  args:
    - run
    - dpll/main.vi
    - "--breadth-first"
---
success: true
exit_code: 0
----- stdout -----
satisfiable
evaluation under model: true
var 1 = true
var 2 = true
var 3 = false
var 4 = true
var 5 = false
var 6 = true
var 8 = false
var 9 = false
var 10 = false
var 11 = false
var 12 = false
var 13 = true
var 14 = true
var 15 = true
var 16 = true
var 17 = true
var 18 = false
var 19 = false
var 20 = false
var 21 = false
var 22 = true
var 23 = true
var 24 = false
var 25 = true
var 26 = false
var 27 = false
var 28 = false
var 29 = true
var 30 = true
var 31 = true
var 32 = false
var 33 = false
var 34 = true
var 35 = true
var 36 = false
var 37 = true
var 38 = true
var 39 = true
var 40 = true
var 41 = true
var 42 = true
var 43 = true
var 44 = true
var 45 = false
var 46 = true
var 47 = false
var 48 = true
var 49 = false
var 50 = false

----- stderr -----

Interactions
  Total            4_911_493
    Depth          2_459_876
    Breadth                1
  Annihilate       2_737_158
  Commute             18_659
  Copy               504_620
  Erase              566_358
  Expand             379_143
  Call               450_068
  Branch             255_487

Memory
  Heap               478_000 B
  Allocated      108_806_032 B
  Freed          108_806_032 B

[Performance...]
